pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -210,6 +219,32 @@ func (c *controller) onEvent(item types.NamespacedName) error {
 	return nil
 }
 
+func (c *controller) onServiceEvent(input any) {
+	event := input.(controllers.Event)
+	curSvc := event.Latest().(*corev1.Service)
+
+	// This is shortcut. We only care about the port number change if we receive service update event.
+	if event.Event == controllers.EventUpdate {
+		oldSvc := event.Old.(*corev1.Service)
+		oldPorts := extractPorts(oldSvc.Spec.Ports)
+		curPorts := extractPorts(curSvc.Spec.Ports)
+		// If the ports don't change, we do nothing.
+		if oldPorts.Equals(curPorts) {
+			return
+		}
+	}
+
+	// We care about add, delete and ports changed event of services that are referred
+	// by ingress using port name.
+	namespacedName := config.NamespacedName(curSvc).String()
```

Comment:
nit: no need to use NamespacedName, can just use Name. We already know they are always in the same namespace